components/calendar/src/cal/hijri.rs

```diff
@@ -837,39 +730,341 @@ impl Calendar for HijriUmmAlQura {
+                PackedHijriYearInfo::new(1581, [l, l, s, l, s, l, s, s, l, s, l, s], fixed_from_iso(2155, 7, 1)),
+                PackedHijriYearInfo::new(1582, [l, l, s, l, l, s, l, s, l, s, s, s], fixed_from_iso(2156, 6, 19)),
+                PackedHijriYearInfo::new(1583, [l, l, s, l, l, l, s, l, s, l, s, s], fixed_from_iso(2157, 6, 8)),
+                PackedHijriYearInfo::new(1584, [s, l, l, s, l, l, s, l, l, s, l, s], fixed_from_iso(2158, 5, 29)),
+                PackedHijriYearInfo::new(1585, [s, l, s, l, s, l, s, l, l, s, l, l], fixed_from_iso(2159, 5, 19)),
+                PackedHijriYearInfo::new(1586, [s, s, l, s, l, s, s, l, l, l, s, l], fixed_from_iso(2160, 5, 8)),
+                PackedHijriYearInfo::new(1587, [s, l, l, s, s, s, l, s, l, s, l, l], fixed_from_iso(2161, 4, 27)),
+                PackedHijriYearInfo::new(1588, [l, s, l, l, s, s, s, l, s, l, s, l], fixed_from_iso(2162, 4, 16)),
+                PackedHijriYearInfo::new(1589, [l, s, l, l, s, l, s, s, l, s, l, s], fixed_from_iso(2163, 4, 5)),
+                PackedHijriYearInfo::new(1590, [l, s, l, l, l, s, s, l, s, l, s, l], fixed_from_iso(2164, 3, 24)),
+                PackedHijriYearInfo::new(1591, [s, l, s, l, l, s, l, s, l, s, l, s], fixed_from_iso(2165, 3, 14)),
+                PackedHijriYearInfo::new(1592, [l, s, l, s, l, s, l, s, l, l, l, s], fixed_from_iso(2166, 3, 3)),
+                PackedHijriYearInfo::new(1593, [l, s, s, l, s, s, l, s, l, l, l, s], fixed_from_iso(2167, 2, 21)),
+                PackedHijriYearInfo::new(1594, [l, l, s, s, l, s, s, s, l, l, l, l], fixed_from_iso(2168, 2, 10)),
+                PackedHijriYearInfo::new(1595, [s, l, s, l, s, s, l, s, s, l, l, l], fixed_from_iso(2169, 1, 30)),
+                PackedHijriYearInfo::new(1596, [s, l, l, s, l, s, s, l, s, l, s, l], fixed_from_iso(2170, 1, 19)),
+                PackedHijriYearInfo::new(1597, [s, l, l, s, l, s, l, s, l, s, l, s], fixed_from_iso(2171, 1, 8)),
+                PackedHijriYearInfo::new(1598, [l, s, l, s, l, l, s, l, s, l, l, s], fixed_from_iso(2171, 12, 28)),
+                PackedHijriYearInfo::new(1599, [s, l, s, l, s, l, s, l, l, l, s, l], fixed_from_iso(2172, 12, 17)),
+                PackedHijriYearInfo::new(1600, [s, s, l, s, l, s, s, l, l, l, s, l], fixed_from_iso(2173, 12, 7)),
+            ]
+        };
+        if let Some(&packed) = usize::try_from(year - 1300).ok().and_then(|i| DATA.get(i)) {
+            HijriYearInfo::unpack(year, packed)
+        } else {
+            HijriYearInfo {
+                value: year,
+                month_lengths: core::array::from_fn(|i| {
+                    HijriTabular::days_in_provided_month(year, i as u8 + 1) == 30
+                }),
+                start_day: calendrical_calculations::islamic::fixed_from_tabular_islamic(
+                    year,
+                    1,
+                    1,
+                    ISLAMIC_EPOCH_FRIDAY,
+                ),
+            }
```

Comment:
Question: Are there tests around the boundary? Sorry if you added this elsewhere but I haven't seen it